sklearn/impute.py

```diff
@@ -548,24 +555,26 @@ class IterativeImputer(BaseEstimator, TransformerMixin):
 
     def __init__(self,
                  missing_values=np.nan,
-                 imputation_order='ascending',
-                 n_iter=10,
                  predictor=None,
                  sample_posterior=False,
+                 max_iter=10,
+                 tol=1e-3,
                  n_nearest_features=None,
                  initial_strategy="mean",
+                 imputation_order='ascending',
                  min_value=None,
                  max_value=None,
                  verbose=0,
                  random_state=None):
 
         self.missing_values = missing_values
-        self.imputation_order = imputation_order
-        self.n_iter = n_iter
         self.predictor = predictor
         self.sample_posterior = sample_posterior
+        self.max_iter = max_iter
+        self.tol = tol
```

Comment:
We should also make a check that the tolerance is positive or 0

Reply:
Is this usually done via an `assert` in the `init`?

Reply:
It is usually done in `fit`. You can do it next to `max_iter`